### ISSUE TITLE
Extension of the maximum "What's new"-tab backlog from 500 to 1000 possible tiles

### ISSUE THREAD

**Issue by Plugdemholes-Imeanpatchdembugs** [TeamNewPipe/NewPipe#11741]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

Hi :)
I've noticed, the first time ever after a months of absence from my tablet that the feed backlog is limited to 500 tiles, so I currently can't see the other two weeks. It's very rare occasion I guess but could the feed be extended to a maximum of 1000 tiles if possible? :)

### Why do you want this feature?

I can't see the rest of the videos that were uploaded in one month of absence, the maximum tiles of 500 "recent" uploads were exceeded. :)

### Additional information

no further remarks

**Comment by Plugdemholes-Imeanpatchdembugs** [comment]
This would be awesome! :)
I myself didn't think I'd ever have to load in so many videos at once. But here I am with my ~170 subscriptions and look and see it's backlog of more than 500. ^^"

**Comment by chiragbharambe** [comment]
Yes, I would also like this feature to be implemented.

**Comment by brwnz** [comment]
> This is the limit you are talking about, I guess:
> 
> https://github.com/TeamNewPipe/NewPipe/blob/14081505cd874117f24c9fce4f8fba2f7031cc09/app/src/main/java/org/schabi/newpipe/database/feed/dao/FeedDAO.kt#L85-L94
> 
> It is there to keep the query fast and memory usage low. I think we could even reduce the limit to ~200 because it is unusual to look further back. Especially because the feed only contains streams that [were published in the last 13 weeks](https://github.com/TeamNewPipe/NewPipe/blob/14081505cd874117f24c9fce4f8fba2f7031cc09/app/src/main/java/org/schabi/newpipe/local/feed/FeedDatabaseManager.kt#L32-L37). **_Instead we could implement lazy loading more items when reaching the end of the displayed list._**

Yes, the 'lazy loading' of new more items would the best/optimal solution. You can than limit the number of initial items, and allow users to  fetch more items when needed.
This would personally help me, as I need to be able to go back and rewatch/find previous videos, but cannot remember which subscription they were in. This is also how it works in FreeTube on GNU/Linux
